Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -1307,6 +1307,50 @@ class CombinatorialKalmanFilter {
       // Emplace back the successful result
       ckfResults.emplace_back(combKalmanResult);
     }
+
+    // Compute nSharedhits and Update ckf results
+    // hit index -> list of multi traj indexes [traj, meas]
+    // @todo put shared hit computation in ad-hoc method
+    std::unordered_map<std::size_t,
+                       std::vector<std::pair<std::size_t, std::size_t>>>
+        recordedHits;
```

Comment:
This data structure probably does a ton of heap allocations. A few ideas would be to:

1. Try to use a *small vector* if possible to save some allocations
2. Try not to use a map but rather a sufficiently sized vector (since the keys should be an integer sequence **I think**
3. Use a pool allocator for both map and vectors.

Reply:
I think it might be better to not check about the shared hits during the track finding. The shared hits info done in this way is not accurate anyway since one CKF invocation only deals with one seed. But it's very likely that the shared hits happen between tracks found via multiple different seeds. 
Hence, a better solution might be that we sort out the shared hits after the all the tracks have been found if necessary. This is more straightforward and accurate.

Reply:
Hi @XiaocongAi, I see however that the `findtrack` method (as called in [TrackFindingAlgorithm.cpp#L68](https://github.com/acts-project/acts/blob/main/Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp#L68)) uses all the initial parameters and measurements. I suppose this is taking into consideration the entire events, right? 

Inside the CombinatorialKalmanFilter.hpp file there is indeed a loop over all the seeds. That is on line [CombinatorialKalmanFilter.hpp#L1265](https://github.com/CarloVarni/acts/blob/main/Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp#L1265 ). However my changes would be after that loop. Shouldn't we have all the tracks at that point, from all the seeds?

Is my understanding right?

We cal always move the computation somewhere else I think to make the code more straigthforward. Would we have all tracks in [TrackFindingAlgorithm.cpp#L89](https://github.com/acts-project/acts/blob/main/Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp#L89)? Or a different (new) algorithm can be added to the sequence.

Reply:
Also, as @paulgessinger suggested, we might use a vector for the `recordedHits` such that the element `i` in the vector would correspond to the info for `hitIndex i` .

Reply:
Hi @XiaocongAi, I have moved the shared hits computation inside `Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/TrackFindingAlgorithm.hpp` in a ad-hoc method.

Also, as @paulgessinger suggested, I have modified the algorithms in order not use `std::unordered_map` but to use two `std::vector<int>` instead.